packages/js/tracks/src/utils.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Internal dependencies
+ */
+import { EVENT_NAME_REGEX, PROP_NAME_REGEX } from './constants';
+import { ExtraProperties } from '.';
+
+export const isDevelopmentMode = process.env.NODE_ENV === 'development';
+
+export function validateEventNameAndProperties(
+	eventName: string,
+	props: ExtraProperties | undefined = {}
+) {
+	let isValid = true;
+	if ( ! EVENT_NAME_REGEX.test( eventName ) ) {
+		if ( isDevelopmentMode ) {
```

Comment:
I added this check here to not show the console error while the tests are being run.

Reply:
I agree, I think validating in unit tests helps because we're adding coverage for tracks in tests as well.